contracts/exchange/mixins/auction/EndemicReserveAuction.sol

```diff
@@ -54,83 +51,71 @@ abstract contract EndemicReserveAuction is
         bytes32 r,
         bytes32 s,
         ReserveAuction calldata auction,
-        ReserveAuction calldata bid
-    ) external onlySupportedERC20Payments(auction.paymentErc20TokenAddress) {
-        if (
-            auction.isBid ||
-            !bid.isBid ||
-            auction.nftContract != bid.nftContract ||
-            auction.tokenId != bid.tokenId ||
-            auction.paymentErc20TokenAddress != bid.paymentErc20TokenAddress ||
-            auction.signer == bid.signer
-        ) revert InvalidConfiguration();
-
-        _verifyApprovalSignature(v, r, s, auction, bid);
-        _verifySignature(auction);
-        _verifySignature(bid);
-
-        AuctionFees memory auctionFees = _calculateAuctionFees(auction, bid);
-
-        if (auction.price + auctionFees.takerCut > bid.price) {
+        ReserveAuction calldata bid,
+        AuctionInfo calldata info
+    ) external onlySupportedERC20Payments(info.paymentErc20TokenAddress) {
+        if (auction.isBid || !bid.isBid || auction.signer == bid.signer) {
```

Comment:
Malo je cudan ovaj konstrukt "bid.isBid". Zasto to imamo na taj nacin rijeseno?

Reply:
Slazem se, izgleda malo cudno. Auction i bid koriste iste strukturu podataka i typehash, oba potpisa se provjeravaju istom funkcijom. Da bi razlikovali kreiranu aukciju i bid, ima ova varijabla "isBid". Mozda mogu napravit da kreirana aukcija i bid imaju razliciti typehash, da se provjeravaju zasebnim funkcijama. Tako bi se onda mogla izbacit ova varijabla "isBid". Ili mozda samo preimenovat da ima vise smisla?

Reply:
Ocemo napravit da imaju onda drugaciji typehash kako bi izbjegli ovaj property?

Reply:
Moze